chart/etcd-backup-restore/templates/etcd-statefulset.yaml

```diff
@@ -162,102 +179,53 @@ spec:
         resources:
 {{ toYaml .Values.resources.backup | indent 10 }}
         env:
+        - name: "POD_NAME"
+          value: {{ .Release.Name }}-etcd-0
+        - name: "POD_NAMESPACE"
+          value: {{ .Release.Namespace }}
         - name: STORAGE_CONTAINER
           value: {{ .Values.backup.storageContainer }}
-{{- if eq .Values.backup.storageProvider "S3" }}
-        - name: "AWS_REGION"
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Release.Name }}-etcd-backup
-              key: "region"
-        - name: "AWS_SECRET_ACCESS_KEY"
+{{- if .Values.backup.storageProvider }}
+  {{- if eq .Values.backup.storageProvider "S3" }}
+        - name: "AWS_APPLICATION_CREDENTIALS"
+          value: "/var/etcd-backup"
+    {{- if .Values.backup.s3.endpoint }}
+        - name: "AWS_ENDPOINT_URL_S3"
           valueFrom:
             secretKeyRef:
               name: {{ .Release.Name }}-etcd-backup
-              key: "secretAccessKey"
-        - name: "AWS_ACCESS_KEY_ID"
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Release.Name }}-etcd-backup
-              key: "accessKeyID"
-{{- else if eq .Values.backup.storageProvider "ABS" }}
-        - name: "STORAGE_ACCOUNT"
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Release.Name }}-etcd-backup
-              key: "storageAccount"
-        - name: "STORAGE_KEY"
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Release.Name }}-etcd-backup
-              key: "storageKey"
-  {{- if .Values.backup.abs.emulatorEnabled }}
+              key: "endpoint"
+              optional: true
+    {{- end }}
+  {{- else if eq .Values.backup.storageProvider "ABS" }}
+        - name: "AZURE_APPLICATION_CREDENTIALS"
+          value: "/var/etcd-backup"
+    {{- if .Values.backup.abs.emulatorEnabled }}
         - name: "AZURE_EMULATOR_ENABLED"
```

Comment:
You can now get rid of `AZURE_EMULATOR_ENABLED` env var, after rebasing on latest master which includes https://github.com/gardener/etcd-backup-restore/pull/819